lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -1483,3 +1550,8 @@ testCertificatePublicationTimes =
 
 instance Arbitrary BlockHeader where
     arbitrary = genSlotNo >>= genBlockHeader
+
+instance Arbitrary POSIXTime where
+    arbitrary = do
+        Positive int <- arbitrary @(Positive Int)
```

Comment:
I think we could avoid this special case of needing a positive integer if we instead use `Maybe` to represent the possibility that there hasn't yet been a GC.

Reply:
How so? Negative values are undefined for POSIXTime. This is unrelated to whether we use Maybe or not.

Reply:
> Negative values are undefined for POSIXTime. This is unrelated to whether we use Maybe or not.

Sorry, I wasn't clear enough here.

I'm referring to the **_zero_** value, not to negative values. The use of `Positive` (in QuickCheck) also prevents the generation of the zero value. (i.e., it's positive in the strict sense.)

Reply:
Both Positive and NonZero are fine here.